### PR TITLE
Task#11

### PR DESCRIPTION
**Pull request by Pimmy17**
Task#11 Create a POST request for a new comment.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -190,4 +190,54 @@ describe('Testing app', () => {
...
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe('Missing Input!')
+            })
+        })
+        test('should return a status error of 400 if invalid info is entered', () => {
+            const anotherBadComment = {
+                article_id: 1,
+                username: 3,
+                body: 'Bob Lob Law',
```
Comment (douglashellowell) [inline __tests__/app.test.js:232]: great content for a law blog

File: __tests__/app.test.js [hunk 2]
```diff
@@ -190,4 +190,54 @@ describe('Testing app', () => {
           .expect(204);
         });
     });
+    describe('POST', () => {
```
Comment (douglashellowell) [inline __tests__/app.test.js:193]: good tests!

consider

`POST /api/articles/9999/comments`
and
`POST /api/articles/not-an-id/comments`

File: app.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ app.use((err, req, res, next) => {
     else next(err)
   })
   
-  app.use((err, rew, res, next) => {
+  app.use((err, req, res, next) => {
```
Comment (douglashellowell) [inline app.js:36]: there are a lot of PSQL errors

you don't need a new error handler for each one!

you can check the `err.code` with if/else/switch blocks in one controller

**Review by douglashellowell — CHANGES_REQUESTED** [review]
Looking great!

There are a few more error tests that could be added

File: __tests__/app.test.js (hunk 1 above)
Comment (Pimmy17) [reply __tests__/app.test.js:232]: I think that's the law firm in Arrested Development!

File: __tests__/app.test.js (hunk 2 above)
Comment (Pimmy17) [reply __tests__/app.test.js:193]: Added

File: app.js (hunk 3 above)
Comment (Pimmy17) [reply app.js:36]: Left this in as I can't find the specific code errors for two of my tests, which fail when I take them out.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -190,4 +190,82 @@ describe('Testing app', () => {
           .expect(204);
         });
     });
+    describe('POST', () => {
+        test('status:201, responds with a new comment added to an article', () => {
+          const testComment = {
+            article_id: 1,
```
Comment (popatre) [inline __tests__/app.test.js:196]: The request body should only accept two things looking at the ticket:
`Request body accepts:

an object with the following properties:
username
body`

The article Id should be handled from the params.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -190,4 +190,82 @@ describe('Testing app', () => {
...
+            .post("/api/articles/1/comments")
+            .send(anotherBadComment)
+            .expect(400)
+            .then(({ body: { msg } }) => {
+              expect(msg).toBe('Incorrect Input!')
+            })
+        });
+        test('should return a status error of 400 if invalid article id is entered', () => {
+            const wrongArticleComment = {
+                article_id: 99999,
```
Comment (popatre) [inline __tests__/app.test.js:244]: Same for these -  you can just put the article_id in the route, and it should be handled from there

File: controllers/articles.controllers.js [hunk 6]
```diff
@@ -44,4 +44,12 @@ exports.deleteComment = (req, res, next) => {
     })
 };
 
-
+exports.addComment = (req, res, next) => {
+    postComment(req.body)
+    .then((comment) => {
```
Comment (popatre) [inline controllers/articles.controllers.js:49]: Need to grab the article_id here from the params.

**Review by popatre — CHANGES_REQUESTED** [review]
Please see the changes regarding article_id

It will be easier when it comes to from end, because you wont have to try and find our the article_id from trying to post a comment. 
Also, the ticket asks for two properties on the post object.

**Pull request merged**